src/builtins.ts

```diff
@@ -1884,16 +1747,26 @@ export function compileCall(
     case BuiltinSymbols.atomic_or:
     case BuiltinSymbols.atomic_xor:
     case BuiltinSymbols.atomic_xchg: {
-      if (!compiler.options.hasFeature(Feature.THREADS)) break;
       if (
+        checkFeatureEnabled(Feature.THREADS, reportNode, compiler) |
         checkTypeRequired(typeArguments, reportNode, compiler, true) |
         checkArgsOptional(operands, 2, 3, reportNode, compiler)
```

Comment:
Could you also add checking type argument is not float (only integers) for atomics?

Reply:
It should do that afterwards, that is emit an `Operation X cannot be applied to type Y` error if there's no matching case. https://github.com/AssemblyScript/assemblyscript/pull/869/files/c01a9c63ffd1d25a24a38cf7eea6c0c7f3f74cea#diff-2c8ebd9fe94e61d77bc89956497d80daR1768

Reply:
Right. [Now I see](https://github.com/AssemblyScript/assemblyscript/pull/869/files/c01a9c63ffd1d25a24a38cf7eea6c0c7f3f74cea#diff-2c8ebd9fe94e61d77bc89956497d80daR1756)